### PR TITLE
feat(components): richlayout with forwardref

### PR DESCRIPTION
**Pull request by jare-talend**
**What is the problem this PR is trying to solve?**

The popover is visible in the list while the popover has the focus.
A button lost his focus if we set it to disabled. So the focus is lost on the popover.

lost_focus

**What is the chosen solution to this problem?**

Add a function to add the focus on the popover

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/RichTooltip/RichLayout/RichLayout.component.js [hunk 1]
```diff
@@ -37,21 +37,32 @@ const Footer = ({ id, children, className }) => (
...
-			<Body id={`${props.id}-body`} className={props.className}>
-				{props.text && <p>{props.text}</p>}
-				{!props.text && Inject.getReactElement(props.getComponent, props.Content)}
-			</Body>
-			<Footer id={`${props.id}-footer`}>
-				{Inject.getReactElement(props.getComponent, props.Footer)}
-			</Footer>
-		</span>
-	);
+export default class RichLayout extends React.Component {
```
Comment (frassinier) [inline packages/components/src/RichTooltip/RichLayout/RichLayout.component.js]: why do you need a class here?

File: packages/components/src/RichTooltip/RichLayout/RichLayout.component.js [hunk 2]
```diff
@@ -37,21 +37,32 @@ const Footer = ({ id, children, className }) => (
...
+		this.richLayout = React.createRef();
+	}
+
+	focusRichLayout() {
+		this.richLayout.current.focus();
+	}
+
+	render() {
+		return (
+			<span id={this.props.id} ref={this.richLayout} tabIndex="-1">
```
Comment (frassinier) [inline packages/components/src/RichTooltip/RichLayout/RichLayout.component.js]: It makes sense here to use FocusTrap I guess

Comment (frassinier) [reply packages/components/src/RichTooltip/RichLayout/RichLayout.component.js]: And do you know why a span was used here? I'm wondering?

File: packages/components/src/RichTooltip/RichLayout/RichLayout.component.js (hunk 1 above)
Comment (jare-talend) [reply packages/components/src/RichTooltip/RichLayout/RichLayout.component.js]: https://fr.reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-class-component.
https://fr.reactjs.org/docs/refs-and-the-dom.html#refs-and-function-components

My components where is set my RichLayout call a focusRichLayout when i lost the focus on the button.

File: packages/components/src/RichTooltip/RichLayout/RichLayout.component.js (hunk 2 above)
Comment (jare-talend) [reply packages/components/src/RichTooltip/RichLayout/RichLayout.component.js]: Legacy I think to wrap the element.

File: packages/components/src/RichTooltip/RichLayout/RichLayout.component.js (hunk 1 above)
Comment (frassinier) [reply packages/components/src/RichTooltip/RichLayout/RichLayout.component.js]: Thank you for the documentation but my comment was more about "why do you change the functional component in favor of a class-based component?"

File: packages/components/src/RichTooltip/RichLayout/RichLayout.component.js (hunk 2 above)
Comment (frassinier) [reply packages/components/src/RichTooltip/RichLayout/RichLayout.component.js]: Could you profit of your PR to change it to a more convenient div, please?

File: packages/components/src/RichTooltip/RichLayout/RichLayout.component.js (hunk 1 above)
Comment (jare-talend) [reply packages/components/src/RichTooltip/RichLayout/RichLayout.component.js]: If i want to access to the method instance, i need a class based component instead a function component.

File: packages/components/src/RichTooltip/RichLayout/RichLayout.component.js (hunk 2 above)
Comment (jare-talend) [reply packages/components/src/RichTooltip/RichLayout/RichLayout.component.js]: done

**Comment by romainseb** [comment]
Same as @jmfrancois could the focusManager make the job here ?
https://github.com/Talend/ui/blob/master/packages/components/src/FocusManager/FocusManager.component.js

**Comment by jare-talend** [comment]
> Same as @jmfrancois could the focusManager make the job here ?
> https://github.com/Talend/ui/blob/master/packages/components/src/FocusManager/FocusManager.component.js

Like react-focus-lock, i need to force the focus on a element, Nope, I try and we lost the focus. We should force the focus on one element.

**Pull request merged**